### PR TITLE
Fix for SR-5615

### PR DESCRIPTION
**Pull request by mackoj**
Change to an error text from **'public' modifier cannot be used in protocols**  to **protocol requirements implicitly have the same access as the protocol**

Resolves [SR-5615](https://bugs.swift.org/browse/SR-5615).

### CODE REVIEW

**Comment by CodaFi** [comment]
This seems like it would be better to have as a note in addition to the error text.

**Comment by mackoj** [comment]
Ok

**Comment by mackoj** [comment]
@CodaFi I have done your modification

File: include/swift/AST/DiagnosticsSema.def [hunk 1]
```diff
@@ -1080,6 +1080,8 @@ ERROR(cdecl_throws,none,
 
 ERROR(attr_methods_only,none,
       "only methods can be declared %0", (DeclAttribute))
+NOTE(access_control_in_protocol_more_detailled_info,none,
+	  "protocol requirements implicitly have the same access as the protocol", ())
```
Comment (xwu) [inline include/swift/AST/DiagnosticsSema.def]: Drive-by nits: "detailed" is misspelled here; also, notes in this file are mostly named "something_note" or "note_something" and not "something_more_detailed_info"; finally, this file uses spaces for alignment--please remove the tab and replace with the appropriate number of spaces.

**Comment by mackoj** [comment]
@xwu I made the changes

File: lib/Sema/TypeCheckAttr.cpp [hunk 2]
```diff
@@ -654,6 +654,7 @@ bool AttributeEarlyChecker::visitAbstractAccessibilityAttr(
 
   // Or within protocols.
   if (isa<ProtocolDecl>(D->getDeclContext())) {
+    diagnoseAndRemoveAttr(attr, diag::access_control_in_protocol_detailed_info);
```
Comment (CodaFi) [inline lib/Sema/TypeCheckAttr.cpp]: You're going to want to emit the note thru the type checker instead of this little convenience function, or you might pop this fixit twice.  Also, emit notes after primary diagnostics.

**Review by CodaFi — CHANGES_REQUESTED** [review]
This requires a test case - you can use the SR if you like.  Please add one in the appropriate place.

File: include/swift/AST/DiagnosticsSema.def [hunk 3]
```diff
@@ -1080,6 +1080,8 @@ ERROR(cdecl_throws,none,
 
 ERROR(attr_methods_only,none,
       "only methods can be declared %0", (DeclAttribute))
+NOTE(access_control_in_protocol_detailed_info,none,
```
Comment (CodaFi) [inline include/swift/AST/DiagnosticsSema.def]: You just seem to have removed the word `more`, which is not quite what @xwu asked.  Take a look at the naming scheme for notes in the rest of this file.

**Comment by mackoj** [comment]
Thanks @CodaFi for the comment I will do the modification you requested

**Comment by CodaFi** [comment]
Superseded by #13486
